design/0003-byop.md

```diff
@@ -0,0 +1,158 @@
+proxies, control plane components, etc. This can be found
+[here](https://github.com/linkerd/linkerd2/blob/master/charts/linkerd2/templates/prometheus.yaml#L26)
+
+Once this is done, Users should be able to see golden metrics, etc in their
+exisiting prometheus instance. The `prometheus.url` field helps users in
+configuring Linkerd components i.e `public-api`,etc that needs prometheus
+access to the exisiting prometheus installation.
+
+Flag interactions are defined below:
+
+| `prometheus.enabled` | `prometheus.url`              | Result                                                                                                                                                                                                                                           |
+|---------|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| true    | ""               | Install linkerd-prometheus, and configure control-plane components to use it.                                                                                                                                                                    |
+| false   | "prometheus.xyz" | No linkerd-prometheus, control-plane components will be configured to url                                                                                                                                                                        |
+| true    | "prometheus.xyz" | Install linkerd-prometheus, but stil configure control plane to use different url, useful when linkerd-prometheus is configured to remote-write into long term storage solution at url, Thus  powering dashboard and CLI through long term storage. |
+| false   | ""               | No linkerd-prometheus, and no URL to power dashboards. (Should we even allow this?)                                                                                                                                                              |
+
+### Linkerd Check Tooling
+
+When users get their own Prometheus, It would be really great to have
+some tooling that allows users to check if `scrape-configs` are configured
+correctly and prometheus is able to get metrics.
+
+`--prometheus` flag can be added to linkerd check to run these tests.
+The following tests are planned:
+
+- Check if the required scrape-configs are present in proemetheus cm.
+( It should be noted that, the prometheus could be present in an differents
+namespace, etc)
+
+- Check if the common metrics are queriable in prometheus, by using the
+end-point i.e `prometheus.url` from the `linkerd-values` cm.
+
+# Prior art
+
+[prior-art]: #prior-art
+
+- We already have docs for [exporting metrics](https://linkerd.io/2/tasks/exporting-metrics/).
+We could update those docs, with remote-write, etc.
```

Comment:
The intent of the Prior Art section is to gather experience from other projects about how they've trying solving a similar problem. For this particular RFC I skimmed what the Prometheus Operator project was doing to parametrize configuration, but that implied setting CRDs and what not, so that was too complex for what we need. Please let us know what you've researched elsewhere, and why/how it could be (or not) applied here.